File: _more/compiler/py2rs/_version/v1/test/fact.c
#include <stdio.h>
#include <stdlib.h>

int factorial(int n) {
    if (n == 0 || n == 1) {
    return 1;
    } else {
    return (n * factorial((n - 1)));
    }
}

int main() {
    int result = factorial(5);
    printf("factorial(5)=%d\n", result);
    return 0;
}

File: _more/compiler/py2rs/_version/v2/test/fact.rs
use std::collections::HashMap;
fn factorial(n: i32) -> i32 {
    if n == 0 || n == 1 {
    return 1;
    } else {
    return (n * factorial((n - 1)));
    }
}

fn main() {
    let result: i32 = factorial(5);
    println!("factorial(5)={}", result);
}